airbyte-integrations/connectors/source-braintree/source_braintree/streams.py

```diff
@@ -0,0 +1,231 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from abc import abstractmethod
+from datetime import datetime
+from typing import Any, Generator, Iterable, List, Mapping, Optional, Union
+
+import backoff
+import braintree
+import pendulum
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.streams.core import Stream
+from braintree.attribute_getter import AttributeGetter
+from source_braintree.schemas import Customer, Discount, Dispute, MerchantAccount, Plan, Subscription, Transaction
+from source_braintree.spec import BraintreeConfig
+
+
+class BraintreeStream(Stream):
```

Comment:
nit: we should declare as ABC since it declares abstract methods
```suggestion
class BraintreeStream(Stream, ABC):
```

Reply:
Updated